### PR TITLE
Tram collision counter hazard lights

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request
To the best of Nanotrasen's ability, valuable crew members are still getting hit by the tram. To further encourage safe practices, a new indicator with hazard lights has been installed outside medbay. It will flash when tram collisions are occurring during times of lax safety for extra visibility.

counterflash

## Why It's Good For The Game
Go for the high score.
## Changelog
:cl: LT3
imageadd: Hazard lights added to the tram collision counter
/:cl:

### CODE REVIEW

**Review by Twaticus — APPROVED** [review]

File: code/game/objects/structures/signs/signs_interactive.dm [hunk 1]
```diff
@@ -111,6 +112,16 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/sign/delamination_counter, 32)
 	SIGNAL_HANDLER
 
 	hit_count++
+	if(hazard_flash)
+		update_appearance()
+		return
+
+	if(hit_count == 11)
```
Comment (Fikou) [inline code/game/objects/structures/signs/signs_interactive.dm:119]: define

File: code/game/objects/structures/signs/signs_interactive.dm [hunk 2]
```diff
@@ -91,6 +91,7 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/sign/delamination_counter, 32)
 	is_editable = TRUE
 	var/hit_count = 0
 	var/tram_id = TRAM_LIFT_ID
+	var/hazard_flash = FALSE
```
Comment (Fikou) [inline code/game/objects/structures/signs/signs_interactive.dm:94]: you need to document this

Comment (lessthnthree) [reply code/game/objects/structures/signs/signs_interactive.dm:95]: Done

File: code/game/objects/structures/signs/signs_interactive.dm [hunk 3]
```diff
@@ -1,3 +1,5 @@
+#define COLLISION_HAZARD_THRESHOLD 11
```
Comment (Fikou) [inline code/game/objects/structures/signs/signs_interactive.dm:1]: you need to undef at the end of the file

**Comment by Jacquerel** [comment]
My one complaint is that the sign is set up so that it looks like it has diagonal hazard lines behind it (drawn on the frame) but the lights light up in a way which separates halves of lines instead of lighting up an entire line at once.

**Comment by lessthnthree** [comment]
> My one complaint is that the sign is set up so that it looks like it has diagonal hazard lines behind it (drawn on the frame) but the lights light up in a way which separates halves of lines instead of lighting up an entire line at once. This emphasises the fact that the bottom right black corner is smaller than the top left one and makes them look like irregularly placed parts of a light tube instead.

Yeah I wasn't totally happy, but the two corners being different is a pain.

What if instead of alternating we had yellow on-off flash? Similar to the pattern below.

image

orangelights2

**Comment by Jacquerel** [comment]
That looks like it could work to me

**Comment by lessthnthree** [comment]
> That looks like it could work to me

I only wanna sprite one of the two options. You like the top or bottom one? I'm leaning bottom. Top one is the current commited one.

**Comment by Jacquerel** [comment]
Oh right sorry I misunderstood, I would probably go with the bottom one.

**Comment by lessthnthree** [comment]
> Oh right sorry I misunderstood, I would probably go with the bottom one.

Good call, it looks better.

**Review by Twaticus — APPROVED** [review]

**Pull request merged**